credentials/production-grid.json

```diff
@@ -4,7 +4,7 @@
   "newscap": "URI:DIR2-RO:4nbaibh3k5fzm56u5dvtttrurq:j2mbmh7p6ax7mgvvah4ugsszqczmphgyfqptiz5ifoxh22i6fhyq",
   "zkap_unit_name": "GB-month",
   "zkap_unit_multiplier": 0.0006,
-  "zkap_payment_url_root": "https://private.storage/payment/",
+  "zkap_payment_url_root": "https://permanent.private.storage/payment/",
```

Comment:
For consistency (with e.g. the `/support` one) it would be best to remove the trailing slashes, IMO.

They could still redirect to "a directory" but my understanding of best-practices here is that not having the slash makes it more correct (i.e. it's "a resource" and its name is `/payment`).

I do like the separate subdomain to remind us  :)

Reply:
> For consistency (with e.g. the /support one) it would be best to remove the trailing slashes, IMO.
> They could still redirect to "a directory" but my understanding of best-practices here is that not having the slash makes it more correct (i.e. it's "a resource" and its name is /payment).

And indeed `/payment` will redirect to `/payment/` for some reason I don't really know but I assume has to do with ... link generation?

Separately, inside S3/CloudFront where the redirect from `permanent...` is handled I _thought_ the query args were being lost from the `/payment` version but not the `/payment/` version but checking it now ... they both work (maybe some change was still propagating when I tested it?  Or I just confused myself).  So, great, I don't think there's a reason to keep the trailing / here.  Thanks!


I agree... but it's complicated :(

`https://private.storage/payment` will redirect to `https://private.storage/payment/` (I'm not sure why we do that but I would guess it has something to do with generating links correctly).

Then somewhere in the S3 or CloudFront

Reply:
Ooohh no.  The query args are lost. :(  The first redirect (from `https://permanent.privatestorage-staging.com/payment?voucher=blub&sha256=blab` to `https://privatestorage-staging.com/333-switch-matomo-domain/payment?voucher=blub&sha256=blab`) preserves them and then the second redirect (from `https://privatestorage-staging.com/333-switch-matomo-domain/payment?voucher=blub&sha256=blab`) drops them (pointing at `https://privatestorage-staging.com/333-switch-matomo-domain/payment/`).

The second redirect is controlled by ... some JavaScript in an AWS Lambda we wrote ... so I guess it is probably technically fixable...

Reply:
They're all `GET`s ...? Or is there a `POST` (which IIRC has different redirect semantics).

Reply:
Anyway, I've _fairly_ confident that `Location:` can include any URI including stuff like query, fragments -- so it **should** be merely a matter of configuration to fix.

Reply:
Anyway, I've _fairly_ confident that `Location:` can include any URI including stuff like query, fragments -- so it **should** be merely a matter of configuration to fix.

Reply:
Yea they're all GETs.  The complication is all just inside AWS and how it handles the requests - not at the plain HTTP level.  I eventually did convince AWS to handle `permanent.private.storage/payment` (and `/payment/` and `/payment?...` and `/payment/?...`) correctly (but don't try `private.storage/payment?...` - that was broken before and I couldn't figure out how to fix it so I didn't).